drivers/console/uart_console.c

```diff
@@ -446,13 +446,13 @@ static void uart_console_isr(const struct device *unused, void *user_data)
 	ARG_UNUSED(user_data);
 	static uint8_t last_char = '\0';
 
-	while (uart_irq_update(uart_console_dev) &&
-	       uart_irq_is_pending(uart_console_dev)) {
+	while (uart_irq_update(uart_console_dev) == 1 &&
```

Comment:
Maybe check > 0 instead? Might be one less instruction...

Reply:
Yep - without any optimization, that saves 2 instructions on RISC-V.